Assignment/CashTest.c

```diff
@@ -0,0 +1,40 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// Vinitha Perumal - GET
+// ------------------------------------------------------------------
+// CashTest.c
+// Program on Test2.2 branch.
+// ------------------------------------------------------------------------------------------------
+#include "Cash.h"
+#include <stdio.h>
+
+void TestCases ();
+
+void TestCases () {
+   int cashPaid[] = { 23, 56, 90 }, actualAmount[] = { 22, 37, 40 },
+      expectedOutput[][4] = { { 0, 0, 0, 1 }, { 1, 1, 2, 0 }, { 5, 0, 0, 0 } },
+      numOfTestCases = sizeof (cashPaid) / sizeof (cashPaid[0]);
+   for (int i = 0; i < numOfTestCases; i++) {
+      printf ("\nTestCase %d\n\nCash Paid: %d\nActual Amount: %d\nExpected Output: 10Rs. %d, "
+              "5Rs. %d, 2Rs. %d, 1Rs. %d\n", i + 1, cashPaid[i], actualAmount[i],
+              expectedOutput[i][0], expectedOutput[i][1], expectedOutput[i][2], expectedOutput[i][3]);
+      int coins[4], res = Cash (cashPaid[i], actualAmount[i], coins), isTestPassed = 1;
+      if (res == 1) {
+         for (int j = 0; j < 4; j++) {
+            if (coins[j] != expectedOutput[i][j]) {
+               isTestPassed = 0;
+               break;
+            }
+         }
+      }
+      else isTestPassed = 0;

```

Comment:
You can simplify this by using [guards](https://en.wikipedia.org/wiki/Guard_(computer_science)).